server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -354,8 +354,11 @@ public ScheduledCancellable schedule(Runnable command, TimeValue delay, String e
     }
 
     public void scheduleUnlessShuttingDown(TimeValue delay, String executor, Runnable command) {
+        if (!Names.SAME.equals(executor)) {
+            command = new ThreadedRunnableAllowShutdown(command, executor(executor));
```

Comment:
what's the reason for this change? It's unrelated to this PR, no?

Reply:
It is related to this PR. If you schedule something on a non-SAME executor and then shutdown the threadpool afterwards (before the delay is passed), the execute on the executor will fail, causing an exception. This exception is thrown on the scheduler thread. So far this caused no issues, since it was ignored (and recently logged). But with the changes in this PR, the exception was bubbled to the uncaught exception handler causing tests to fail.

Above fixes that such that `scheduleUnlessShuttingDown` allows both the schedule call itself to pass and the subsequent `execute` on the executor.

Reply:
Will the regular `.schedule` method not suffer from the same issue?

Reply:
Yes, it will potentially, though none of the CI runs ran into it.

I guess we could argue that before the warning/exception changes, this would go unnoticed for both `schedule` and `scheduleUnlessShuttingDown` and therefore we should silently ignore this in both cases?

Reply:
I think the shutdown procedure is relevant to this question. `ThreadPool.terminate` does a regular shutdown first on both scheduler and pool executors. This allows any scheduled tasks on the `ScheduledThreadPoolExecutor` to run to completion but does not allow executing them on the pool executor.

So scheduling jobs using SAME executor with delay until after shutdown will complete, using any other executor will not.

Given that we have both methods, I think the right solution is to make sure that tasks scheduled to run within the terminate timeout will run. I think this belongs in a follow-up PR. Two options:

1. During terminate, we can do awaitTermination of the scheduler before calling shutdown on the rest of the executors.
2. If a scheduled job fails to call execute on the executor, we simply call it in current thread (ie. one of the scheduler threads).